src/libcephfs.cc

```diff
@@ -322,7 +322,15 @@ struct ceph_mount_info
 
   int chdir(const char *to, const UserPerm& perms)
   {
-    return client->chdir(to, perms);
+    int rc = client->chdir(to, perms);
+    if (rc == 0) {
+      /* Current API requires "cwd" to be refreshed after every chdir so that
+       * getcwd on an unlinked cwd will still return the old path. Note:
+       * Client::getcwd now returns an error but leaves the "cwd" string
```

Comment:
OK. So now I read this comment back-referring to getcwd() returning -ENOENT. While I'm fine with the refactor, why is it an optimization?

Reply:
Well, the optimization was to stop doing a pointless `Client::getcwd` with every `::chdir` but apparently that was for libcephfs's benefit for a really subtle corner case. So I standby that I "optimized" the `Client` and pushed the corner-case handling into `libcephfs` where it belongs.